### ISSUE TITLE
feat: add differential privacy training — DP-SGD with tensorflow-privacy

### ISSUE THREAD

**Issue by ShaistaShabbir-prog** [ShaistaShabbir-prog/AIX#9]
## Feature: Differential Privacy for LSTM training

Extend the privacy work by training WITH differential privacy and comparing MIA success rate.

### Hypothesis
A model trained with DP-SGD should have lower MIA accuracy than a standard model.

### Implementation
```python
from tensorflow_privacy.optimizers.dp_optimizer_keras import DPKerasSGDOptimizer

dp_optimizer = DPKerasSGDOptimizer(
    l2_norm_clip=1.0,
    noise_multiplier=0.1,
    num_microbatches=256,
    learning_rate=0.01
)
```

### Experiment
- Train: Standard LSTM vs DP-LSTM
- Attack: MIA on both
- Compare: attack accuracy should drop with DP
- Privacy budget (ε) analysis
